src/gui/models/nodePaletteFilterProxy.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "gui/models/nodePaletteFilterProxy.h"
+#include "procedure/nodes/registry.h"
+
+// Set current context
+void NodePaletteFilterProxy::setContext(ProcedureNode::NodeContext context)
+{
+    context_ = context;
+    invalidateFilter();
+}
+
+/*
+ * QSortFilterProxyModel overrides
+ */
+
+bool NodePaletteFilterProxy::filterAcceptsRow(int row, const QModelIndex &parent) const
+{
+    if (context_ == ProcedureNode::NodeContext::AnyContext)
+        return true;
+
+    static const std::map<ProcedureNode::NodeContext, std::vector<std::string>> contextCategories = {
+        {ProcedureNode::NodeContext::NoContext, {}},
+        {ProcedureNode::NodeContext::AnalysisContext, {"Data"}},
+        {ProcedureNode::NodeContext::GenerationContext,
+         {"Calculate", "Build", "Calculate", "Data", "General", "Operate", "Pick", "Potentials", "Sites"}},
+        {ProcedureNode::NodeContext::OperateContext,
+         {"Calculate", "Data", "General", "Operate", "Pick", "Potentials", "Sites"}},
+    };
+
+    const auto &contextCategoryList = contextCategories.at(context_);
+    auto category = std::next(ProcedureNodeRegistry::categoryMap().begin(), parent.isValid() ? parent.row() : row)->first;
+    return std::find(contextCategoryList.begin(), contextCategoryList.end(), category) != contextCategoryList.end();
```

Comment:
I'm finding it hard to decide whether this is they way we should do it or not!  Given that, for a given context, I think there are either very few, or very many, exclusions to make, perhaps it might be better to structure it something like this:
```suggestion
    auto category = std::next(ProcedureNodeRegistry::categoryMap().begin(), parent.isValid() ? parent.row() : row)->first;
    
    switch (context_)
    {
        case (ProcedureNode::NodeContext::AnyContext):
            return true;
        case (ProcedureNode::NodeContext::NoContext):
            return false;
        case (ProcedureNode::NodeContext::OperateContext):
            return category == "Operate";
        case (ProcedureNode::NodeContext::GenerateContext):
            return category != "Operate";
        case (ProcedureNode::NodeContext::AnalysisContext):
            return category != "Build" && category != "Potentials" && category != "Operate";
        default:
            throw(.....);
    }
```

Reply:
That's certainly nicer than trying to maintain quite a large map - much less likely to miss something out!